src/app/clusters/scenes-server/scenes-server.cpp

```diff
@@ -0,0 +1,940 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "scenes-server.h"
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/CommandHandlerInterface.h>
+#include <app/InteractionModelEngine.h>
+#include <app/clusters/scenes-server/SceneTableImpl.h>
+#include <app/server/Server.h>
+#include <app/util/attribute-storage.h>
+#include <credentials/GroupDataProvider.h>
+#include <lib/support/CommonIterator.h>
+#include <lib/support/Span.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <platform/PlatformManager.h>
+
+using SceneTableEntry   = chip::scenes::DefaultSceneTableImpl::SceneTableEntry;
+using SceneStorageId    = chip::scenes::DefaultSceneTableImpl::SceneStorageId;
+using SceneData         = chip::scenes::DefaultSceneTableImpl::SceneData;
+using HandlerContext    = chip::app::CommandHandlerInterface::HandlerContext;
+using ExtensionFieldSet = chip::scenes::ExtensionFieldSet;
+using GroupDataProvider = chip::Credentials::GroupDataProvider;
+using SceneTable        = chip::scenes::SceneTable<chip::scenes::ExtensionFieldSetsImpl>;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace Scenes {
+
+ScenesServer ScenesServer::mInstance;
+
+ScenesServer & ScenesServer::Instance()
+{
+    return mInstance;
+}
+
+CHIP_ERROR ScenesServer::Init()
+{
+    // Prevents re-initializing
+    VerifyOrReturnError(!mIsInitialized, CHIP_ERROR_INCORRECT_STATE);
+
+    ReturnErrorOnFailure(chip::app::InteractionModelEngine::GetInstance()->RegisterCommandHandler(this));
+    VerifyOrReturnError(registerAttributeAccessOverride(this), CHIP_ERROR_INCORRECT_STATE);
+
+    mSceneTable = chip::scenes::GetSceneTableImpl();
+    ReturnErrorOnFailure(mSceneTable->Init(&chip::Server::GetInstance().GetPersistentStorage()));
+
+    mGroupProvider = Credentials::GetGroupDataProvider();
+
+    mFeatureFlags.Set(ScenesFeature::kSceneNames);
+    // The bit of 7 the NameSupport attribute indicates whether or not scene names are supported
+    //
+    // According to spec, bit 7 (Scene Names) MUST match feature bit 0 (Scene Names)
+    mNameSupport = 0x80;
+
+    // TODO: Make modifications to the CommandHandler and Transport layer to be able to forward the NodeId to clusters
```

Comment:
`GetSubjectDescriptor()` on the command handler followed by checking that authMode is `AuthMode::kCase` and then getting the subject will give you the information you want, no?

Reply:
I missed that mechanism. Todo removed, will be added in next commit when the attribute dirty marking and endpoint scoping is properly implemented.